#!/usr/bin/env python
PACKAGE = 'zero'

from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add('coe1',  double_t,    0, 'coe of yaw',    0.15,    -10, 5)
gen.add('coe2',  double_t,    0, 'coe of pitch',               0.0015,      -10, 5)
gen.add('KEY', int_t,    0, '0-picture;1-video',1 ,0,1)
gen.add('kk',  double_t,    0, 'Depends on m(kg)',    0.12,    0, 3)
gen.add('dt', double_t,    0, '1/FPS',               0.1,   0, 0.4)
gen.add('l',  double_t,    0, 'NOISE',               1,      0, 300)
gen.add('u',  double_t,    0, 'NOISE',               1,      0, 10)
gen.add('v',  double_t,    0, 'VELOCITY',            10,     0, 30)
gen.add('test_pattern',  bool_t,    0, 'test_pattern',True)
gen.add('coe_of_yaw', double_t,    0, 'coe_of_yaw',               1,   0, 2)
gen.add('coe_of_pitch', double_t,    0, 'coe_of_pitch',               1,   0, 2)
gen.add('yaw',  double_t,    0, 'yaw',    0,    -360, 360)
gen.add('pitch', double_t,    0, 'pitch',   20,   -20, 70)
gen.add('pid_dt', double_t,    0, 'pid_dt',   0.15,   0, 1)
gen.add('pid_max', double_t,    0, 'pid_max',   10,   0, 360)
gen.add('pid_min', double_t,    0, 'pid_min',   -10,   -360, 360)
gen.add('pid_Kp', double_t,    0, 'pid_Kp',   0,   0, 2)
gen.add('pid_Ki', double_t,    0, 'pid_Ki',   0.,   0, 1)
gen.add('pid_Kd', double_t,    0, 'pid_Kd',   0,   0, 1)


exit(gen.generate(PACKAGE, 'zero','cv'))
